src/python/pants/backend/python/rules/pex_from_target_closure.py

```diff
@@ -37,7 +37,7 @@ class CreatePexFromTargetClosure:
   all_target_adaptors = [t.adaptor for t in all_targets]
 
   interpreter_constraints = PexInterpreterConstraints.create_from_adaptors(
-    adaptors=tuple(all_targets),
+    adaptors=tuple(all_target_adaptors),
```

Comment:
This was a bug in production. We would ignore the `compatibility` field for targets and always use the global `--python-setup-interpreter-constraints`. We never use `compatibility` anymore internally, so we didn't detect this.